### PR TITLE
Fix WorkflowStub#cancel and WorkflowStub#terminate exceptions contract

### LINKED ISSUE #1198: WorkflowStub#cancel and #terminate throw StatusRuntimeException instead of WorkflowNotFoundException
## Expected Behavior

If a workflow doesn't exist, `WorkflowStub#cancel` and `WorkflowStub#terminate` should throw a `WorkflowNotFoundException` like other methods of `WorkflowStub` like `#signal`.

## Actual Behavior

Raw gRPC `StatusRuntimeException` with NOT_FOUND code is thrown.

### PR DESCRIPTION
**Pull request by Spikhalskiy**
Fix WorkflowStub#cancel and WorkflowStub#terminate exceptions contract
Align behavior of test server with the real one in some edge cases related to cancelations.

Closes #1198

### CODE REVIEW

File: temporal-sdk/src/main/java/io/temporal/client/WorkflowStub.java [hunk 1]
```diff
@@ -97,32 +135,88 @@ static <T> WorkflowStub fromTyped(T typed) {
...
+   *     availability issues
    */
   <R> R getResult(long timeout, TimeUnit unit, Class<R> resultClass, Type resultType)
       throws TimeoutException;
 
   /**
-   * Returns workflow result potentially waiting for workflow to complete. Behind the scene this
-   * call performs long poll on Temporal service waiting for workflow completion notification.
+   * Returns a {@link CompletableFuture} with the workflow result potentially waiting for workflow
+   * to complete. Behind the scene this call performs long poll on Temporal service waiting for
```
Comment (bergundy) [inline temporal-sdk/src/main/java/io/temporal/client/WorkflowStub.java:157]:
```suggestion
   * to complete. Behind the scenes this call long polls the Temporal service for 
```

**Review by bergundy — APPROVED** [review]
